src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -0,0 +1,2526 @@
+            //                    }
+            //                    DiagnosticUtility.TraceHandledException(e, TraceEventType.Information);
+            //                }
+            //                finally
+            //                {
+            //                    if (threwException)
+            //                    {
+            //                        this.channelBinder.Abort();
+            //                    }
+            //                }
+            //            }
+
+
+            //static void ReceiveCallback(IAsyncResult result)
+            //{
+            //    ((SessionInitiationMessageHandler)result.AsyncState).ProcessMessage(result);
+            //}
+        }
+
+        interface IInputQueueChannelAcceptor
+        {
+            int PendingCount { get; }
+        }
+
+        interface IServerSecuritySessionChannel
+        {
+            void RenewSessionToken(SecurityContextSecurityToken newToken, SecurityContextSecurityToken supportingToken);
+        }
+
+        abstract class ServerSecuritySessionChannel : /*ChannelBase,*/ IServerSecuritySessionChannel
+        {
+            FaultCode renewFaultCode;
+            FaultReason renewFaultReason;
+            FaultCode sessionAbortedFaultCode;
+            FaultReason sessionAbortedFaultReason;
+
+            // Double-checked locking pattern requires volatile for read/write synchronization
+            volatile bool areFaultCodesInitialized;
+            IServerReliableChannelBinder channelBinder;
+            SecurityProtocol securityProtocol;
+            // This is used to sign outgoing messages
+            SecurityContextSecurityToken currentSessionToken;
+            UniqueId sessionId;
+            // These are renewed tokens that have not been used as yet
+            List<SecurityContextSecurityToken> futureSessionTokens;
+            SecuritySessionServerSettings settings;
+            RequestContext initialRequestContext;
+            volatile bool isInputClosed;
+            // ThreadNeutralSemaphore receiveLock;
+            MessageVersion messageVersion;
+            SecurityListenerSettingsLifetimeManager settingsLifetimeManager;
+            volatile bool hasSecurityStateReference;
+            EndpointAddress localAddress;
+
+            protected ServerSecuritySessionChannel(SecuritySessionServerSettings settings,
+                // IServerReliableChannelBinder channelBinder,
+                SecurityContextSecurityToken sessionToken,
+                object listenerSecurityProtocolState,
+                SecurityListenerSettingsLifetimeManager settingsLifetimeManager, EndpointAddress address)
+            // : base(settings.securityServiceDispatcher)
+            {
+                this.settings = settings;
+                // this.channelBinder = channelBinder;
+                this.messageVersion = settings.MessageVersion;
+                // this.channelBinder.Faulted += this.OnInnerFaulted;
```

Comment:
Does it make sense for this class to Fault now? If it does still make sense, then need to wire something up here sensible.

Reply:
Created issue, let's discuss the need (since channelBinder work is reduced) https://github.com/CoreWCF/CoreWCF/issues/263